lib/livebook/intellisense.ex

```diff
@@ -519,14 +522,33 @@ defmodule Livebook.Intellisense do
     """
   end
 
+  defp format_definition_link(module, function_or_type \\ nil) do
+    path =
+      :code.get_path()
+      |> Enum.find(&(to_string(&1) =~ "livebook_runtime"))
+      |> to_string()
+      |> Path.join("#{module}.beam")
+
+    if File.exists?(path) do
+      query =
+        case function_or_type do
+          {:function, fun, arity} ->
+            %{module: module_name(module), function: fun, arity: arity}
+
+          {:type, type, arity} ->
+            %{module: module_name(module), type: type, arity: arity}
+
+          nil ->
+            %{module: module_name(module)}
+        end
+
+      "[Go to definition](./go-to-definition?#{URI.encode_query(query)})"
```

Comment:
Today this link will be [hijacked](https://github.com/livebook-dev/livebook/blame/main/assets/js/lib/markdown.js#L259-L262) and converted into a push. But it is not really a link in practice. I wonder if instead we could do this:

```suggestion
      "[Go to definition](goto:definition?#{URI.encode_query(query)})"
```

Then, in the JavaScript code I linked above, you intercept "goto:" links and convert them into a `pushEvent` to the server. This way, there is no navigation, no need to pass baseUrl around, etc.

Reply:
I'll check that

Reply:
Actually, I am almost sure this is the case, which is why we have `LIVEBOOK_ALLOW_URI_SCHEMES` to allow some extra schemes.